pallets/parachain-system/src/lib.rs

```diff
@@ -971,6 +973,40 @@ impl<T: Config> GetChannelInfo for Pallet<T> {
 }
 
 impl<T: Config> Pallet<T> {
+	/// Updates inherent data to only contain messages that weren't already processed
+	/// by the runtime based on MQC heads.
+	///
+	/// # Panics
+	///
+	/// Panics if MQC heads don't match even after exhausting incoming message queue, either
```

Comment:
This function doesn't panic, docs here are referring to the panic in `set_validation_data`?

Reply:
Leftover from previous version of PR, my bad